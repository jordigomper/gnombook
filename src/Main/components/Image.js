import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import styled from "@emotion/styled";

const Img = styled.img`
  object-fit: cover;
  height: 100px;
`;

const EmptyBackground = styled.img`
  height: 100px;
  background-color: gray;
`;

const ImageCache = ({ src, alt, title }) => {
  const [image, setImage] = useState();

  useEffect(() => {
    const img = new Image();
    img.onload = function() {
      setImage(img.src);
    };
    img.src = src;
  }, []);

  return !image ? (
    <EmptyBackground />
  ) : (
    <Img src={image} alt={alt} title={title} />
  );
};

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
};

Image.defaultProps = {
  src: null,
  alt: null,
  title: null
};

export default ImageCache;
