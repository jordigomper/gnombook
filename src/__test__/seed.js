import React, { useContext, createContext } from "react";

export const useAppContext = () => useContext(AppContext);
const defaultValues = [];
const AppContext = createContext(defaultValues);
export default AppContext;
