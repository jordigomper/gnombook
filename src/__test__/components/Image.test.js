import React from "react";
import { shallow } from "enzyme";
import Image from "../../Main/components/Image";
import * as AppContext from "../seed";

describe("Image component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Image />);
    expect(wrapper).toMatchSnapshot();
  });
});
