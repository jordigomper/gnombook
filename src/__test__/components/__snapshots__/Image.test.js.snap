// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component starts with a count of 0 1`] = `ShallowWrapper {}`;

exports[`Image component renders correctly 1`] = `ShallowWrapper {}`;

exports[`Image component starts with a count of 0 1`] = `ShallowWrapper {}`;
